[2024-12-05T17:26:58.841+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-12-05T17:26:58.849+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Weather_Realtime_Data_ETL.load_weather_current_data scheduled__2024-11-28T02:00:00+00:00 [queued]>
[2024-12-05T17:26:58.853+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Weather_Realtime_Data_ETL.load_weather_current_data scheduled__2024-11-28T02:00:00+00:00 [queued]>
[2024-12-05T17:26:58.853+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-12-05T17:26:58.859+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): load_weather_current_data> on 2024-11-28 02:00:00+00:00
[2024-12-05T17:26:58.864+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=254) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-05T17:26:58.863+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Weather_Realtime_Data_ETL', 'load_weather_current_data', 'scheduled__2024-11-28T02:00:00+00:00', '--job-id', '444', '--raw', '--subdir', 'DAGS_FOLDER/ETL_weather_realtime.py', '--cfg-path', '/tmp/tmpz87znbor']
[2024-12-05T17:26:58.865+0000] {standard_task_runner.py:63} INFO - Started process 260 to run task
[2024-12-05T17:26:58.865+0000] {standard_task_runner.py:91} INFO - Job 444: Subtask load_weather_current_data
[2024-12-05T17:26:58.891+0000] {task_command.py:426} INFO - Running <TaskInstance: Weather_Realtime_Data_ETL.load_weather_current_data scheduled__2024-11-28T02:00:00+00:00 [running]> on host 5de3b6ca18a7
[2024-12-05T17:26:58.974+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='shatayu.thakur@sjsu.edu' AIRFLOW_CTX_DAG_OWNER='Shatayu' AIRFLOW_CTX_DAG_ID='Weather_Realtime_Data_ETL' AIRFLOW_CTX_TASK_ID='load_weather_current_data' AIRFLOW_CTX_EXECUTION_DATE='2024-11-28T02:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-28T02:00:00+00:00'
[2024-12-05T17:26:58.975+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-12-05T17:26:58.980+0000] {base.py:84} INFO - Using connection ID 'snowflake_conn' for task execution.
[2024-12-05T17:26:58.981+0000] {connection.py:399} INFO - Snowflake Connector for Python Version: 3.10.0, Python Version: 3.12.3, Platform: Linux-6.10.11-linuxkit-aarch64-with-glibc2.36
[2024-12-05T17:26:58.982+0000] {connection.py:1239} INFO - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
[2024-12-05T17:26:59.885+0000] {cursor.py:1149} INFO - Number of results in first chunk: 1
[2024-12-05T17:26:59.890+0000] {logging_mixin.py:188} INFO - Realtime Record to be pushed to snowflake table:                   date  temp  temp_feel  humidity  rain  year  month  day  hour
0 2024-12-05 12:15:00  39.9       29.7        53   0.0  2024     12    5    12
[2024-12-05T17:27:00.056+0000] {cursor.py:1149} INFO - Number of results in first chunk: 1
[2024-12-05T17:27:00.272+0000] {cursor.py:1149} INFO - Number of results in first chunk: 1
[2024-12-05T17:27:00.634+0000] {cursor.py:1149} INFO - Number of results in first chunk: 1
[2024-12-05T17:27:00.634+0000] {logging_mixin.py:188} INFO - Target Table Initialized and Ready to Store Realtime Weather Data from ETL
[2024-12-05T17:27:01.136+0000] {logging_mixin.py:188} INFO - Inserted current weather for date: 2024-12-05 12:15:00
[2024-12-05T17:27:01.136+0000] {logging_mixin.py:188} INFO - Inserted current weather record successfully to dev.raw_data.weather_current through query: 
            MERGE INTO dev.raw_data.weather_current AS target
            USING (SELECT '2024-12-05 12:15:00' AS date, 39.9 AS temp, 29.7 AS temp_feel, 
                        53 AS humidity, 0.0 AS rain, 2024 AS year, 
                        12 AS month, 5 AS day, 12 AS hour) AS source
            ON target.date = source.date
            WHEN MATCHED THEN 
                UPDATE SET target.temp = source.temp, target.temp_feel = source.temp_feel, 
                        target.humidity = source.humidity, target.rain = source.rain, 
                        target.year = source.year, target.month = source.month, 
                        target.day = source.day, target.hour = source.hour
            WHEN NOT MATCHED THEN
                INSERT (date, temp, temp_feel, humidity, rain, year, month, day, hour) 
                VALUES (source.date, source.temp, source.temp_feel, source.humidity, 
                        source.rain, source.year, source.month, source.day, source.hour);
            
[2024-12-05T17:27:01.253+0000] {cursor.py:1149} INFO - Number of results in first chunk: 1
[2024-12-05T17:27:01.253+0000] {logging_mixin.py:188} INFO - Transaction committed. Data load complete.
[2024-12-05T17:27:01.254+0000] {connection.py:762} INFO - closed
[2024-12-05T17:27:01.292+0000] {connection.py:768} INFO - No async queries seem to be running, deleting session
[2024-12-05T17:27:01.362+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-12-05T17:27:01.363+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-12-05T17:27:01.367+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=Weather_Realtime_Data_ETL, task_id=load_weather_current_data, run_id=scheduled__2024-11-28T02:00:00+00:00, execution_date=20241128T020000, start_date=20241205T172658, end_date=20241205T172701
[2024-12-05T17:27:01.408+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-12-05T17:27:01.415+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-05T17:27:01.416+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
